### PR TITLE
Handling callback event and fetching message status by id

### PR DESCRIPTION
**Pull request by ankita10r**
@avikganguly01 
@SubhamPramanik : please review

### CODE REVIEW

**Comment by ankita10r** [comment]
@SubhamPramanik : the code changes in this PR pertain to callback event that would be triggered after the provider has successfully delivered a sms. [this is an update on the previous merged PR https://github.com/openMF/message-gateway/pull/37]
Code changes also includes provision for manually fetching the message status from provider if the callback does not happen.

File: src/main/java/org/fineract/messagegateway/sms/api/SmsApiResource.java [hunk 1]
```diff
@@ -42,9 +52,22 @@
...
 	//This class sends TRANSACTIONAL & PROMOTIONAL SMS
+
 	private SMSMessageService smsMessageService ;
+
 	private static final Logger logger = LoggerFactory.getLogger(SmsApiResource.class);
 
+	@Autowired
+	private TelerivetMessageProvider telerivetMessageProvider;
+
+	private SmsOutboundMessageRepository smsOutboundMessageRepository;
```
Comment (SubhamPramanik) [inline src/main/java/org/fineract/messagegateway/sms/api/SmsApiResource.java]: How is the object created? Should it be auto wired?

Comment (ankita10r) [reply src/main/java/org/fineract/messagegateway/sms/api/SmsApiResource.java]: this is a redundant  object declaration. I missed to remove it as it is not serving any purpose now

Comment (SubhamPramanik) [reply src/main/java/org/fineract/messagegateway/sms/api/SmsApiResource.java]: So can you remove it then?

Comment (ankita10r) [reply src/main/java/org/fineract/messagegateway/sms/api/SmsApiResource.java]: yeah just updating this PR itself

Comment (SubhamPramanik) [reply src/main/java/org/fineract/messagegateway/sms/api/SmsApiResource.java]: Yes. Thank you.

**Pull request merged**